jupyter_server_fileid/manager.py

```diff
@@ -588,16 +580,16 @@ def _sync_file(self, path, stat_info):
             return None
 
         src = self.con.execute(
-            "SELECT id, path, crtime, mtime FROM Files WHERE ino = ?", (stat_info.ino,)
+            "SELECT id, path, crtime FROM Files WHERE ino = ?", (stat_info.ino,)
         ).fetchone()
 
         # if ino is not in database, return None
         if src is None:
             return None
-        id, old_path, crtime, mtime = src
+        id, old_path, crtime = src
 
         # if timestamps don't match, delete existing record and return None
-        if not self._check_timestamps(stat_info, crtime, mtime):
+        if crtime != stat_info.crtime:
```

Comment:
Probably an extreme edge case, but if the filesystem is upgraded from one that doesn't support `crtime` to one that does (or vice versa for that matter), this will result in the incorrect deletion of the ID.  You could probably address this by ensuring both operands have values...
```suggestion
        if crtime and stat_info.crtime and crtime != stat_info.crtime:
```

I suspect the check below at L749 would require similar protection.

Reply:
"Upgrading a filesystem" isn't really possible; it's more like copying your files from an old filesystem to a new filesystem. AFAIK there's no way to upgrade a filesystem while guaranteeing preservation of all inode numbers per file. So File ID would break in this edge case regardless.

Reply:
True.